src/applications/mhv-secure-messaging/components/shared/SmRouteNavigationGuard.jsx

```diff
@@ -43,12 +44,56 @@ const SmRouteNavigationGuard = ({
     [isBlocking, setIsBlocking, updateModalVisible],
   );
 
+  const localStorageValues = useMemo(() => {
+    return {
+      atExpires: localStorage.atExpires,
+      hasSession: localStorage.hasSession,
+      sessionExpiration: localStorage.sessionExpiration,
+      userFirstName: localStorage.userFirstName,
```

Comment:
@KolbyVA great question. From what I recall, it is not that these values being undefined that signs you out. Rather the opposite - when a user is signed out (by explicit action or timeout) then these values will be removed from localStorage. The reason why these lines of code are her is to handle the `onbeforeunload` when a user clicks on `Sign out` button. We were noticing back when we incorporated this concept to a Compose page, that this button removes these values prematurely. Meaning even if we intercept navigate away with `beforeunload` handler, those values get removed. To prevent that, we are reintroducing them back in this form. Although there may be potentially a better way of handling it, we were not able to come up with one yet